extensions/blocks/contact-form/components/jetpack-contact-form.js

```diff
@@ -190,6 +200,57 @@ class JetpackContactForm extends Component {
 		);
 	}
 
+	renderOnSubmissionFields() {
+		const { instanceId } = this.props;
+		const {
+			customThankyou,
+			customThankyouType,
+			customThankyouMessage,
+			customThankyouRedirect,
+		} = this.props.attributes;
+		return (
+			<Fragment>
+				<ToggleControl
+					label={ __( 'Show a thank you message', 'jetpack' ) }
+					checked={ customThankyou }
+					onChange={ value => this.props.setAttributes( { customThankyou: value } ) }
+				/>
+				{ customThankyou && (
+					<SelectControl
+						label={ __( 'Message type', 'jetpack' ) }
+						value={ customThankyouType }
+						options={ [
+							{ label: __( 'Text message', 'jetpack' ), value: 'message' },
+							{ label: __( 'Redirect to another webpage', 'jetpack' ), value: 'redirect' },
+						] }
+						onChange={ value => this.props.setAttributes( { customThankyouType: value } ) }
+					/>
+				) }
+				{ customThankyou && ( ! customThankyouType || 'message' === customThankyouType ) && (
+					<TextareaControl
+						label={ __( 'Message text', 'jetpack' ) }
+						value={ customThankyouMessage }
+						placeholder={ __( 'Thank you for your submission!', 'jetpack' ) }
+						onChange={ value => this.props.setAttributes( { customThankyouMessage: value } ) }
+					/>
+				) }
+				{ customThankyou && 'redirect' === customThankyouType && (
+					<BaseControl
```

Comment:
Having to use `BaseControl` here is a little janky, but it's the easiest way to add label in an appropriately styled and accessible manner. WordPress/gutenberg#15669 would soft deprecate the need for this workaround.

Reply:
Just for funsies, 290863f51 can be applied whenever WordPress/gutenberg#15669 becomes available to use.

Reply:
The above could make a good inline-code comment. :-)

Reply:
Another way to go about this URLInput is to wrap it into a component so that it can be used in other places where we ask for URLs (in this form block, contact info block?)

Reply:
Is there a particular format y'all prefer for tracking "this hack can be undone once WP x.y is the minimum we support"?

I don't know that it's worth maintaining a separate component, particularly given that `URLInput` is looking like it's going to get some attention in Core.

Reply:
> Is there a particular format y'all prefer for tracking "this hack can be undone once WP x.y is the minimum we support"?

You can use a comment starting with `@todo`, and mentioning `WP 5.3`. That should help us find it when the time comes. 👍